### PR TITLE
automatically run `regenerate_filters.sh` on pull request

### PR DESCRIPTION
**Pull request by FaustVX**
Hi Liam,

I just wanted to automatized your repo slightly, for your nice project.

It will generate the `LIST` file whenever the `PLAN_LIST` or `regenerate_filters.sh` (to add new search engine) is modified on a pull request.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -15,8 +15,10 @@ There's also the [PLAIN_LIST](https://github.com/LiamDawe/steamdeck_spamblock/bl
 # Contributing
 
 * To have a site accepted in a pull request, please list examples of their spamminess.
-* Please add the site to the PLAIN_LIST as well in your PR.
-* If able, please run the `./regenerate_filters.sh` script to update the LIST file with filters for the newly added site.
+* Please add the site to the `PLAIN_LIST` as well in your PR.
+* Please don't edit the `LIST` file manualy, as it is automaticaly generated.
+* If tou know what you are doing, edit `regenerate_filters.sh
```
Comment (LiamDawe) [inline README.md]: There’s a small spelling error here. Would you mind editing? :)

Comment (FaustVX) [reply README.md]: Yes, it's resolved (I think, I'm not a native English speaker)

**Comment by LiamDawe** [comment]
Hey, this is really cool!

Is there a safe way for me to test it?

**Comment by FaustVX** [comment]
> Is there a safe way for me to test it?

Yes, edit the `PLAN_LIST` on a new branch, then create a pull request, then watch the new commit being made in the `actions` tabs, and then close the PR and delete the newly-created branch.

Or create a private repo to test it (that's what I did)

**Comment by LiamDawe** [comment]
Awesome, thanks :)

**Comment by LiamDawe** [comment]
Now it's merged, do I need to add this anywhere in the repo settings or is it all automatic?

**Comment by FaustVX** [comment]
> do I need to add this anywhere in the repo settings or is it all automatic?

It's totally automatic. You have nothing to do.

**Comment by FaustVX** [comment]
@LiamDawe 
I see you added manually a new site.
To have it automatically, you have to create a pull request.

**Comment by LiamDawe** [comment]
Hmm, I just tested adding a new one via pull request, and the LIST did not get updated.

**Comment by FaustVX** [comment]
OK, I understand what's wrong, I've misnamed the workflow file.
Could you rename the `.github/workflows/generate_links.sh` to `.github/workflows/generate_links.yaml` (https://github.com/LiamDawe/steamdeck_spamblock/edit/main/.github/workflows/generate_links.sh)
It will work fine after that.

Copy/pasting can be dangerous sometime :)

It work fine in my repo after renamed it (https://github.com/FaustVX/steamdeck_spamblock/pull/2)

**Comment by LiamDawe** [comment]
Thanks, let's see how that goes then :)

**Comment by FaustVX** [comment]
I hope it's good now

**Comment by LiamDawe** [comment]
It worked now, thanks :D

**Pull request merged**